lib/jsdom/living/nodes/HTMLFormElement-impl.js

```diff
@@ -122,6 +125,20 @@ class HTMLFormElementImpl extends HTMLElementImpl {
   set action(V) {
     this.setAttribute("action", V);
   }
+
+  // https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#statically-validate-the-constraints
+  checkValidity() {
+    const controls = descendantsByHTMLLocalNames(this, submittableElements);
+    // Even though one negative result would be enough, as per spec the checkValidity code needs to run for all
+    // elements
+    const invalidControls = controls.filter(control => !control.checkValidity());
+    return invalidControls.length === 0;
```

Comment:
Done.

> This does not fire any invalid events, or handle the result of firing them.
Since `.checkValidity()` was used, that one fires the invalid events
> Are there no tests for these missing steps? :(
There are tests for `invalid` events being fired, I don't think there are tests for form within a form however (so that form owner would of the descendant would not be the farther form).